app/views/govuk_publishing_components/components/docs/radio.yml

```diff
@@ -52,6 +52,29 @@ examples:
         text: "Use Government Gateway"
       - value: "govuk-verify"
         text: "Use GOV.UK Verify"
+  with_legend:
+    description: This adds a `legend` element containing the text
```

Comment:
I'd suggest putting this next to the heading example, as they're linked and we want people to find them. Might even be worth combining the two into the `with_legend` example, with a description like:

```
This adds a legend element containing the text supplied.

By default, this will render as a H2 inside the legend. This can be changed to a different heading level using the heading_level option.

If a legend without a heading is required, set heading_level to 0.
```

Reply:
I'm also wondering about replacing most of the places where we use `heading` with `legend`, as that's more accurate.

Reply:
I thought this also, do you think it's worth thinking about changing the element as well:

```
heading: "What's it to do with?"
heading_level: 0
```

```
legend: "What's it to do with?"
heading_level: 0
```

Reply:
It's probably worth doing, but it would introduce a breaking change and make this whole thing more complicated. Up to you.

Reply:
My stomach is saying - I'll note but leave this one for the moment for when there's a bit more confidence with this type change.  However I'll update the examples.

Reply:
> I'm also wondering about replacing most of the places where we use heading with legend, as that's more accurate.

It's semi accurate, it's _nested_ headings in this instance. I think I'm going to mentioned this rather than change them all to legends